wayland-client/src/event_queue.rs

```diff
@@ -723,6 +723,8 @@ fn early_callback<
     let (proxy, event) = I::parse_event(handle, msg)?;
     if let Some(proxy_data) = proxy.data() {
         <State as Dispatch<I, U, State>>::event(data, &proxy, event, proxy_data, handle, qhandle);
+    } else {
+        crate::log_debug!("Ignoring event on {}, user_data was never set", I::interface().name);
```

Comment:
This should be *at least* a `log_warning` imo.

Reply:
It's possible to hit this harmlessly in error conditions (failure to construct a proxy).  With #531 that becomes less of an issue and it may be worth upgrading in severity - or the entire thing becomes moot if dropped.